server/service/src/asset/validate.rs

```diff
@@ -27,3 +28,40 @@ pub fn check_asset_log_exists(
 pub fn check_user_is_user(ctx: &ServiceContext, input: &InsertAssetLog) -> bool {
     return ctx.user_id == input.user_id;
 }
+
+pub fn check_reason_matches_status(status: &Option<Status>, reason: &Option<Reason>) -> bool {
+    return match status {
```

Comment:
This match statement felt a bit clumsy. I wondered if there was a better way to ensure reason matching status. Especially if reasons or status change or are dynamic? But I wasn't sure

Reply:
here's my suggestion - the None cases both return true, so I think get that out of the way first. 

```
pub fn check_reason_matches_status(status: &Option<Status>, reason: &Option<Reason>) -> bool {
    let status = match status {
        Some(status) => status,
        None => return true,
    };

    let reason = match reason {
        Some(reason) => reason.to_owned(),
        None => return true,
    };

    match status {
        Status::NotInUse => {
            reason == Reason::AwaitingDecomissioning
                || reason == Reason::Stored
                || reason == Reason::OffsiteForRepairs
                || reason == Reason::AwaitingInstallation
        }

        Status::FunctioningButNeedsAttention => {
            reason == Reason::NeedsServicing || reason == Reason::MultipleTemperatureBreaches
        }
        Status::NotFunctioning => {
            reason == Reason::Unknown
                || reason == Reason::NeedsSpareParts
                || reason == Reason::LackOfPower
        }

        _ => false,
    }
}
```